src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -410,8 +413,33 @@ public class GenerateProtoTask extends DefaultTask {
       }
     }
 
-    cmd.addAll protoFiles
-    logger.log(LogLevel.INFO, cmd.toString())
+    int lengthLimit = getCmdLengthLimit()
+
+    String baseCmd = cmd.join(" ") + " "
+
+    StringBuilder currentCmd = new StringBuilder(baseCmd)
```

Comment:
I think a unit test for this logic is necessary. To make this testable, consider defining a static method like this:
```java
List<String> generateCmds(String baseCmd, List<File> protoFiles, int cmdLineLimit)
```